### PR TITLE
Pywps 4.2.1 implementation

### PR DESCRIPTION
**Pull request by jorgejesus**
# Overview

Implementation of pywps 4.2.1  Ticket #49 
It implements docker images that were tested with new pywps 4.2.1

# Related Issue / Discussion

The current dockerfiles point to the pywps_4.2 branch, this is a sort of chickend and egg problems since docker images point to pywps-flask and from there to the new pywps version

requirements have been change to point to pywps number version tags

# Additional Information

If PR is approved it is necessary to path the dockerfiles on master to point to the proper locations. 
With PR approval it will be see the automatic build in dockerhub and

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [ ] I'd like to contribute [feature X|bugfix Y|docs|something else] to PyWPS. I confirm that my contributions to PyWPS will be compatible with the PyWPS license guidelines at the time of contribution.
- [x ] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,100 +1,134 @@
...
+
+And to run it::
+
+    $ docker run -p 8081:8081 -it pywps/gunicorn-alpine:latest
+
+or::
+ 
+    $ docker run -e GU_WORKERS=10 -e GU_PORT=8082  -p 8082:8082 -it pywps/gunicorn-alpine:latest
+
+Pywps will be available in  the following URL::
```
Comment (cehbrecht) [inline README.rst]: "... *at* the following URL". 

There are some more typos in the text.

**Comment by cehbrecht** [comment]
@jorgejesus I personally prefer to use conda environments instead of virtualenv. You can also install `gdal` using conda. If you like I can add the "conda way" as follow-up of this PR.

File: README.rst [hunk 2]
```diff
@@ -103,11 +137,77 @@ In this case pywps (only the WPS) will be avalable on::
...
+
+
+Nginx-Ubuntu (production)
+-------------------------
+
+This image is based on ``Flask-Ubuntu`` and will require it (either build locally or pull from dockerhub). This image has Nginx and Gunicorn totally integrated as services in a docker image::
+
+
+   $ cd docker/ubuntu/nginx/Dockerfile
+   $ docker build -t pywps/nginx-ubuntu
```
Comment (cehbrecht) [inline README.rst]: "." missing in docker build

File: README.rst [hunk 3]
```diff
@@ -103,11 +137,77 @@ In this case pywps (only the WPS) will be avalable on::
...
+    $ docker run -p 5000:5000 pywps/flask-ubuntu
+
+
+Nginx-Ubuntu (production)
+-------------------------
+
+This image is based on ``Flask-Ubuntu`` and will require it (either build locally or pull from dockerhub). This image has Nginx and Gunicorn totally integrated as services in a docker image::
+
+
+   $ cd docker/ubuntu/nginx/Dockerfile
```
Comment (cehbrecht) [inline README.rst]: skip `/Dockerfile`

File: README.rst [hunk 4]
```diff
@@ -103,11 +137,77 @@ In this case pywps (only the WPS) will be avalable on::
     http://localhost
 
 
+Flask-Ubuntu (basic)
+--------------------
+
+The same as ``Flask-Ubuntu`` but using phusion image (ubuntu 18.04)::
+
+
+    $ cd docker/ubuntu/flask/Dockerfile
```
Comment (cehbrecht) [inline README.rst]: skip `/Dockerfile`

**Comment by cehbrecht** [comment]
... building `gdal` from source takes long when building the docker images. Using conda for docker builds could be an alternative ... maybe for later.

File: requirements.txt [hunk 5]
```diff
@@ -9,4 +9,4 @@ Shapely
 Werkzeug
 SQLAlchemy
 psutil
--e git+https://github.com/geopython/pywps.git@master#egg=pywps-master
+-e git+https://github.com/geopython/pywps.git@4.2.1#egg=pywps-4.2.1
```
[the remark this answers is not in the document]
Comment (jorgejesus) [reply requirements.txt]: Changed to pywps package, tested and it is working

**Pull request merged**